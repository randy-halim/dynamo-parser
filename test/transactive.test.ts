// import Item, { Transactive } from '../src/index';
// import z from 'zod';
// import DocumentClient from './document-client.config';

// const $Book = new Item(
//   'my-table',
//   {
//     id: z.string().max(5),
//     bookName: z.string(),
//     rating: z
//       .number()
//       .min(1)
//       .max(5),
//   },
//   ['id'],
//   DocumentClient
// );

describe('Transactive Get Operations', () => {
  it('Need to make tests', () => {
    expect(true).toBe(true);
  });
});
