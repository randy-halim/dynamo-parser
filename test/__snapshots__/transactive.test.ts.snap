// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Transactive Get Operations Gets all items (Books) 1`] = `
Array [
  Object {
    "bookName": "The Earth Tome",
    "id": "BOOK1",
    "rating": 4,
  },
  Object {
    "bookName": "The Water Tome",
    "id": "BOOK2",
    "rating": 5,
  },
  Object {
    "bookName": "The Fire Tome",
    "id": "BOOK3",
    "rating": 5,
  },
]
`;

exports[`Transactive Get Operations Should throw an error getting a non-existent item 1`] = `
[Error: [
  {
    "code": "invalid_type",
    "expected": "object",
    "received": "undefined",
    "path": [],
    "message": "Required"
  }
]]
`;
